docs/custom-domain/registry-setup.md

```diff
@@ -0,0 +1,82 @@
+ ACR supports both user assigned and system assigned MI. If you choose to enable user assigned MI for your registry, create one first before proceeding. Instructions [here](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal).
+
+### Enable preview features
+1. `az login`
+2. `az account set -s <subscription-id-or-name> `
+3. `apiVersion="api-version=2019-12-01-preview"`
+4. `mgmtEndpoint=$(az cloud show --query endpoints.resourceManager -o tsv)`
+5. `accessToken=$(az account get-access-token --query accessToken -o tsv)`
+6. `registryResourceId=$(az acr show -n myregistry --query id -o tsv)`
+7. You can either enable a system assigned MI, or a user assigned MI, or both for your registry. Do one of the following:
+   - System assigned: 
+     - `identity="{\"type\": \"systemAssigned\"}"`
+   - User assigned: 
+     - `userMIResourceId=$(az identity show -n <user-mi-name> -g <user-mi-resource-group> --subscription <user-mi-subscription-id-or-name> --query id -o tsv)`
+     - `identity="{\"type\": \"userAssigned\", \"userAssignedIdentities\": {\"${userMIResourceId}\":{}}}"`
+   - Both:
+     - `userMIResourceId=$(az identity show -n <user-mi-name> -g <user-mi-resource-group> --subscription <user-mi-subscription-id-or-name> --query id -o tsv)`
+     - `identity="{\"type\": \"systemAssigned,userAssigned\", \"userAssignedIdentities\": {\"${userMIResourceId}\":{}}}"`
+8. `payload="{\"identity\": $identity, \"properties\": {\"dataEndpointEnabled\": true}}"`
+9. `endpoint="$mgmtEndpoint$registryResourceId?$apiVersion"`
+10. `curl -f -X PATCH -H "Authorization: Bearer $accessToken" -H "Content-Type: application/json" $endpoint -d "$payload"`
+
+## Prepare your Azure Key Vault
+For each domain, its TLS certificate must be added to an Azure Key Vault that is accessible by your registry. We recommend creating a new key vault containing only your TLS certificates and granting the registry's identity access to `get` secret.
+1. [Create](https://docs.microsoft.com/en-us/azure/key-vault/) a new Azure Key Vault.
+2. [Add](https://docs.microsoft.com/en-us/azure/key-vault/certificate-scenarios) your certificates to the key vault.
+3. Add an access policy to the key vault that grants your registry's identity access to `get` secret:\
+   `az keyvault set-policy --name <your-kv-name> --secret-permissions get --spn <registry-system-or-user-mi-client-id>`
+   - The output of the command to enable preview features on the registry will contain these client ids.
+
+For greater isolation, you may choose to put each certificate in its own key vault and set its access policy independently. The registry should have always have access to the TLS certificates.
```

Comment:
you may choose --> it is recommended
should have always have --> repeating words.

Reply:
Updated